drivers/SmartThings/matter-sensor/src/air-quality-sensor/init.lua

```diff
@@ -398,14 +396,19 @@ local function measurementHandlerFactory(capability_name, attribute, target_unit
       device:set_field(capability_name.."_unit", reporting_unit, {persist = true})
     end
 
+    local value, err_msg
     if reporting_unit then
-      local value = unit_conversion(ib.data.value, reporting_unit, target_unit)
-      device:emit_event_for_endpoint(ib.endpoint_id, attribute({value = value, unit = unit_strings[target_unit]}))
+      value, err_msg = unit_conversion(ib.data.value, reporting_unit, target_unit)
+    end
 
+    if value then
+      device:emit_event_for_endpoint(ib.endpoint_id, attribute({value = value, unit = unit_strings[target_unit]}))
       -- handle case where device profile supports both fineDustLevel and dustLevel
       if capability_name == capabilities.fineDustSensor.NAME and device:supports_capability(capabilities.dustSensor) then
         device:emit_event_for_endpoint(ib.endpoint_id, capabilities.dustSensor.fineDustLevel({value = value, unit = unit_strings[target_unit]}))
       end
+    else
+      log.info_with({hub_logs = true}, err_msg)
```

Comment:
Should probably be warn level since to call more attention to it

Reply:
Done in [602bced](https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/pull/1582/commits/602bced7425981a5551e35879da067bf0545b606)